stdlib/public/core/Integers.swift.gyb

```diff
@@ -605,59 +648,66 @@ public func &${x.operator} <T: FixedWidthInteger>(lhs: T, rhs: T) -> T {
 //===--- UnsignedInteger --------------------------------------------------===//
 //===----------------------------------------------------------------------===//
 
-public protocol UnsignedInteger_ : BinaryInteger {
+/// This protocol is an implementation detail of `UnsignedInteger`;
+/// do not use it directly.
+@_show_in_interface
+public protocol _DisallowMixedSignArithmetic : BinaryInteger {
+  // Used to create a deliberate ambiguity in cases like UInt(1) +
+  // Int(1), which would otherwise compile due to the arithmetic
+  // operators defined for Strideable types (unsigned types are
+  // Strideable).
+  associatedtype _DisallowMixedSignArithmetic : SignedInteger = Int
+}
+
+public protocol UnsignedInteger : _DisallowMixedSignArithmetic, BinaryInteger {
   associatedtype Magnitude : BinaryInteger
 }
 
-extension UnsignedInteger_ {
+extension UnsignedInteger {
   @_transparent
   public var magnitude: Self { return self }
 
   @_transparent
   public static var isSigned: Bool { return false }
 
   public var description: String {
-    // FIXME(integers): uncomment once Int conforms to BinaryInteger
-    fatalError()
-    //if self == 0 {
-      //return "0"
-    //}
+    if self == 0 {
+      return "0"
+    }
 
-    //let ascii0 = 48
-    //var buf: [UnicodeScalar] = []
+    let ascii0 = 48
+    var buf: [UnicodeScalar] = []
 
-    //var x = self
-    //repeat {
-      //let r = x % 10
-      //x /= 10
-      //buf.append(
-        //UnicodeScalar(
-          //ascii0 + Int(Word(extendingOrTruncating: r)._storage)))
-    //}
-    //while x != 0
-    //return String(buf.reversed().lazy.map { Character($0) })
+    var x = self
+    repeat {
+      let r = x % 10
+      x /= 10
+      buf.append(
+        UnicodeScalar(
+          ascii0 + Int(UInt(extendingOrTruncating: r)._value))!)
+    }
+    while x != 0
+    return String(buf.reversed().lazy.map { Character($0) })
   }
 }
 
-extension UnsignedInteger_ where Self : FixedWidthInteger {
+extension UnsignedInteger where Self : FixedWidthInteger {
   @_transparent
   public init<T : BinaryInteger>(_ source: T) {
     // FIXME(integers): uncomment checks
-    //_assertCond(
-      //source >= 0, "negative value \(source) not representable by \(Self.self)")
+    //_precondition(source >= 0,
```

Comment:
The old checks provide better user experience and I *think* _assertCond might be something the performance team can optimize better.  We should check with them before making the change to _precondition.